elastic/elastic-agent
# [Feature] Start/stop monitoring server based on monitoring config

## What does this PR do?

Adds a capability to turn off monitoring server in case monitoring metrics is disabled. 

## Why is it important?

Up until now monitoring server was always turned on listening on a monitoring port even though monitoring was not enabled. 

TODO: check with cloud they have monitoring enabled in a config

## Checklist


## How to test
- install standalone agent
- play with `agent.monitoring.enabled` and `agent.monitoring.metrics` flags
- any of them set to false should stop server
- verify server is in desired state running `sudo lsof -i :6791`. for server stopped no output, for server running LISTEN and metricbeat connecting to it

Fixes #2734
